pkg/util/resource/manager.go

```diff
@@ -0,0 +1,47 @@
+package resource
+
+import (
+	"github.com/authgear/authgear-server/pkg/api/apierrors"
+)
+
+var ErrResourceNotFound = apierrors.NotFound.WithReason("ResourceNotFound").
+	New("specified resource is not configured")
+
+type Manager struct {
+	Registry *Registry
+	Fs       []Fs
+}
+
+func NewManager(registry *Registry, fs []Fs) *Manager {
+	return &Manager{Registry: registry, Fs: fs}
+}
+
+func (m *Manager) Read(desc Descriptor, args map[string]interface{}) (*LayerFile, error) {
+	var layers []LayerFile
+	for _, fs := range m.Fs {
+		files, err := desc.ReadResource(fs)
+		if err != nil {
+			return nil, err
+		}
+		layers = append(layers, files...)
+	}
+	if len(layers) == 0 {
+		return nil, ErrResourceNotFound
+	}
+
+	merged, err := desc.Merge(layers, args)
```

Comment:
When do we call Parse?

Reply:
Parse is used in resolver to parse templates & translation map.